src/modules/Popup/Popup.js

```diff
@@ -341,6 +344,21 @@ export default class Popup extends Component {
     this.setPopupStyle()
   }
 
+  handleTriggerRef = (triggerRef) => {
+    debug('triggerMounted()')
+    if (triggerRef) {
+      this.triggerRef = triggerRef
+    }
+  }
+
+  setPosition = () => {
+    if (this.triggerRef) {
+      this.setPopupStyle(this.props.position)
+    }
+
+    this.animationRequestId = requestAnimationFrame(this.setPosition)
```

Comment:
haven't you introduced an infinite loop now? When does this ever get short circuited?

Reply:
good point.  fixed in 7c2104f

Reply:
Unfortunately the infinite loop is still there - all the time the popup is displayed `setPosition()` is called in a loop, it then calls `setPopupStyle` which re-sets component's state which causes re-render - takes ~20% CPU when the popup is displayed.